service/src/main/java/com/zebrunner/reporting/service/TestRunService.java

```diff
@@ -288,22 +289,17 @@ public TestRun updateTestRun(TestRun testRun) {
     }
 
     @Transactional
-    public Set<TestRunArtifact> attachTestRunArtifacts(Set<TestRunArtifact> testRunArtifacts, Long id) {
-        Set<TestRunArtifact> attachedArtifacts = new HashSet<>();
+    public List<TestRunArtifact> attachTestRunArtifacts(List<TestRunArtifact> testRunArtifacts, Long id) {
```

Comment:
Why would you switch from set to list everywhere?

Reply:
test artifact not overrides equals and hashcode methods. It is not mandatory to use list instead of set